src/services/ui/src/components/RHF/utils/validator.ts

```diff
@@ -0,0 +1,255 @@
+      return message;
+    }
+  }
+  // TODO: Add validate condition
+  // if (rules?.validate) {
+  //   if (isFunction(rules?.validate)) {
+  //     const result = await rules?.validate(inputValue, formValues);
+  //     const validateError = getValidateError(result, inputRef);
+
+  //     if (validateError) {
+  //       error[name] = {
+  //         ...validateError,
+  //         ...appendErrorsCurry(
+  //           INPUT_VALIDATION_RULES.validate,
+  //           validateError.message
+  //         ),
+  //       };
+  //       if (!validateAllFieldCriteria) {
+  //         setCustomValidity(validateError.message);
+  //         return error;
+  //       }
+  //     }
+  //   } else if (isObject(validate)) {
+  //     let validationResult = {} as FieldError;
+
+  //     for (const key in validate) {
+  //       if (!isEmptyObject(validationResult) && !validateAllFieldCriteria) {
+  //         break;
+  //       }
+
+  //       const validateError = getValidateError(
+  //         await validate[key](inputValue, formValues),
+  //         inputRef,
+  //         key
+  //       );
+
+  //       if (validateError) {
+  //         validationResult = {
+  //           ...validateError,
+  //           ...appendErrorsCurry(key, validateError.message),
+  //         };
+
+  //         setCustomValidity(validateError.message);
+
+  //         if (validateAllFieldCriteria) {
+  //           error[name] = validationResult;
+  //         }
+  //       }
+  //     }
+
+  //     if (!isEmptyObject(validationResult)) {
+  //       error[name] = {
+  //         ref: inputRef,
+  //         ...validationResult,
+  //       };
+  //       if (!validateAllFieldCriteria) {
+  //         return error;
+  //       }
+  //     }
+  //   }
+  // }
+
+  // If all checks pass, the input value is valid
+  return "";
+};
+
+export const validateOption = (optionValue: string, options: any[]) => {
+  return options.find((OPT: any) => OPT.value === optionValue);
+};
+
+export const formGroupValidator =
+  (data: any) => (ACC: ERROR, FORM: T.FormGroup) => {
+    FORM.slots.reduce(slotValidator(data), ACC);
+    return ACC;
+  };
+
+export const slotValidator =
+  (data: any) =>
+  (ACC: ERROR, SLOT: T.RHFSlotProps): ERROR => {
+    const optionValidator = (OPT: T.RHFOption) => {
+      if (OPT.form) OPT.form.reduce(formGroupValidator(data), ACC);
+      if (OPT.slots) {
+        OPT.slots.reduce(slotValidator(data), ACC);
+      }
+      return ACC;
+    };
+
+    const fieldValidator = (FLD: any) => (SLOT1: T.RHFSlotProps) => {
+      if (SLOT1.rhf === "FieldArray") {
+        FLD[SLOT1.name].forEach((DAT: any) => {
+          SLOT1.fields?.forEach(fieldValidator(DAT));
+        });
+      } else if (SLOT1.rhf === "FieldGroup") {
+        FLD[SLOT1.name].forEach((DAT: any) => {
+          SLOT1.fields?.forEach(fieldValidator(DAT));
+        });
+      } else {
+        slotValidator(FLD)(ACC, SLOT1);
+      }
+    };
+
+    if (SLOT.rhf === "Input") {
```

Comment:
SLOTs can only be one of the rhf types, any reason this can't be a switch, or at least an if/else tree to lower amount of checks?

Reply:
I try to avoid switches, because it's always felt too much "tool" for the job.  In addition, not every developer is familiar with the rules of using a "switch".  Even I occasionally forget to break after every case statement. 

As for `if/else`, it's mostly a personal preference.  For me, I lean towards conditional logic that are simpler with less mental overhead.  So, the closer the logic can be expressed in a flat list structure, the easier it'll be to read.